lib/src/working_copy.rs

```diff
@@ -489,6 +509,13 @@ impl TreeState {
                     continue;
                 }
                 let sub_path = dir.join(&RepoPathComponent::from(name));
+                if let Some(file_state) = self.file_states.get(&sub_path) {
+                    if matches!(file_state.file_type, FileType::GitSubmodule{ .. }) {
+                        deleted_files.remove(&sub_path);
+                        continue;
```

Comment:
Could you add a test case or two in `test_working_copy.rs`? It would be good to test that files and directories at the submodule path are not add to the tree when you snapshot the working copy. There is an example in that file of how you can fake a submodule by referring to a commit in the same repo. You can copy how that's done.

Reply:
I added a new test `test_gitsubmodule` in `test_working_copy.rs` it only tests for the git backend as I don't know what the native backend is and how/if it should deal with submodules. I also had to fix `test_checkout_file_transitions` by creating a directory for the submodule to accommodate the changes in this PR